esp-hal/src/lcd_cam/lcd/i8080.rs

```diff
@@ -450,51 +354,127 @@ impl<'d, CH: DmaChannel, DM: Mode> I8080<'d, CH, DM> {
             w.lcd_update().set_bit();
             w.lcd_start().set_bit()
         });
+
+        Ok(I8080Transfer {
+            i8080: self,
+            tx_buf: data,
+        })
     }
+}
 
-    fn tear_down_send(&mut self) {
-        // This will already be cleared unless the user is trying to cancel,
-        // which is why this is still here.
-        self.lcd_cam
+impl<'d, CH: DmaChannel, DM: Mode> core::fmt::Debug for I8080<'d, CH, DM> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.debug_struct("I8080").finish()
+    }
+}
+
+/// Represents an ongoing (or potentially finished) transfer using the I8080 LCD
+/// interface
+pub struct I8080Transfer<'d, CH: DmaChannel, DM: Mode, BUF> {
+    i8080: I8080<'d, CH, DM>,
+    tx_buf: BUF,
+}
+
+impl<'d, CH: DmaChannel, DM: Mode, BUF> I8080Transfer<'d, CH, DM, BUF> {
+    /// Returns true when [Self::wait] will not block.
+    pub fn is_done(&self) -> bool {
+        self.i8080
+            .lcd_cam
             .lcd_user()
-            .modify(|_, w| w.lcd_start().clear_bit());
+            .read()
+            .lcd_start()
+            .bit_is_clear()
+    }
 
-        self.lcd_cam
+    /// Stops this transfer on the spot and returns the peripheral and buffer.
+    pub fn cancel(mut self) -> (I8080<'d, CH, DM>, BUF) {
+        self.stop_peripherals();
+        let (_, i8080, buf) = self.wait();
+        (i8080, buf)
+    }
+
+    /// Waits for the transfer to finish and returns the peripheral and buffer.
+    ///
+    /// Note: This also clears the transfer interrupt so it can be used in
+    /// interrupt handlers to "handle" the interrupt.
+    pub fn wait(self) -> (Result<(), DmaError>, I8080<'d, CH, DM>, BUF) {
+        while !self.is_done() {}
+
+        // Clear "done" interrupt.
+        self.i8080
+            .lcd_cam
             .lc_dma_int_clr()
             .write(|w| w.lcd_trans_done_int_clr().set_bit());
+
+        unsafe {
+            let i8080 = core::ptr::read(&self.i8080);
```

Comment:
Same here, please use `ManuallyDrop`. I'm trying to think how we could formulate this in a nicer way, but until then I'd like to ask for semantic clarity.